velox/exec/tests/utils/Cursor.cpp

```diff
@@ -125,113 +125,242 @@ bool TaskQueue::hasNext() {
   return !queue_.empty();
 }
 
-std::atomic<int32_t> TaskCursor::serial_;
-
-TaskCursor::TaskCursor(const CursorParameters& params)
-    : maxDrivers_{params.maxDrivers},
-      numConcurrentSplitGroups_{params.numConcurrentSplitGroups},
-      numSplitGroups_{params.numSplitGroups} {
-  std::shared_ptr<core::QueryCtx> queryCtx;
-  if (params.queryCtx) {
-    queryCtx = params.queryCtx;
-  } else {
-    // NOTE: the destructor of 'executor_' will wait for all the async task
-    // activities to finish on TaskCursor destruction.
-    executor_ = std::make_shared<folly::CPUThreadPoolExecutor>(
-        std::thread::hardware_concurrency());
-    static std::atomic<uint64_t> cursorQueryId{0};
-    queryCtx = std::make_shared<core::QueryCtx>(
-        executor_.get(),
-        core::QueryConfig({}),
-        std::unordered_map<std::string, std::shared_ptr<Config>>{},
-        cache::AsyncDataCache::getInstance(),
-        nullptr,
-        nullptr,
-        fmt::format("TaskCursorQuery_{}", cursorQueryId++));
+class TaskCursorBase : public TaskCursor {
+ public:
+  explicit TaskCursorBase(const CursorParameters& params) {
+    if (params.queryCtx) {
+      queryCtx_ = params.queryCtx;
+    } else {
+      // NOTE: the destructor of 'executor_' will wait for all the async task
+      // activities to finish on TaskCursor destruction.
+      executor_ = std::make_shared<folly::CPUThreadPoolExecutor>(
+          std::thread::hardware_concurrency());
+      static std::atomic<uint64_t> cursorQueryId{0};
+      queryCtx_ = std::make_shared<core::QueryCtx>(
+          executor_.get(),
+          core::QueryConfig({}),
+          std::unordered_map<std::string, std::shared_ptr<Config>>{},
+          cache::AsyncDataCache::getInstance(),
+          nullptr,
+          nullptr,
+          fmt::format("SingleThreadedTaskCursorQuery_{}", cursorQueryId++));
+    }
+
+    if (!params.queryConfigs.empty()) {
+      auto configCopy = params.queryConfigs;
+      queryCtx_->testingOverrideConfigUnsafe(std::move(configCopy));
+    }
+
+    planFragment_ = {
+        params.planNode,
+        params.executionStrategy,
+        params.numSplitGroups,
+        params.groupedExecutionLeafNodeIds};
+    const std::string taskId =
+        fmt::format("test_single_threaded_cursor {}", ++serial_);
```

Comment:
serial_ and cursorQueryId represent similar things (atomic counters used to generate unique task and query IDs), but names are very different. Would be nice to use consistent naming. Also, can serial_ be a local variable just like cursorQueryId?